### DISCUSSION TITLE
Signout function in V2 issue

### DISCUSSION THREAD

**Discussion by brentrobbins** [nhost/nhost#96] [Q&A]
Hello. I am currently having issues with the `nhost.auth.signOut()` function in V2. It does logout/signout the user, but doesn't kick them out to the login page (like the same setup did in V1). I am using the exact same function along with the same Next.js protected route setup in the docs. After triggering the sign out function the page breaks and triggers errors since they are no longer signed in, however the private route (that is wrapping that page) is not triggered/runs so doesn't redirect the user out to the login page. However when I refresh the page, it then routes me to the login page triggering the private route component.

**Comment by elitan** [comment]
This sounds very much like it relates to this issue: https://github.com/nhost/nhost/discussions/97

Do you use our Next.js template? If not, could you either provide some code or add a minimal reproducible example in a public GitHub repo for us to debug?

I'll also take an extra look at our current Next.js template with SDKs to make sure they work.

**Reply by brentrobbins** [reply]
@elitan I think you might be right that it relates to that issue. 

I am pretty sure when I tried the current Next.js template it didn't work to connect until I updated the packages. 

Here's some of my code that should help.

Private Route: 
```
import { useRouter } from 'next/router'
import { useNhostAuth } from '@nhost/react-auth'

export function authProtected(Comp) {
  return function AuthProtected(props) {
    const router = useRouter()
    const { isLoading, isAuthenticated } = useNhostAuth()

    if (isLoading) {
      return <div>Loading...</div>
    }

    if (!isAuthenticated) {
      router.push('/login')
      return null
    }

    return <Comp {...props} />
  };
}
```

Pages (protected) example:
```
import { authProtected } from '@/components/auth/private-route'

function AppDashboardPage() {
  return (
   ...
  )
}

export default authProtected(AppDashboardPage)
```

Sign Out link/button:
```
 <button onClick={() => nhost.auth.signOut()} className='block px-4 py-2 text-sm text-gray-700'>Sign Out</button>
```
